cypress/e2e/permit2.test.ts

```diff
@@ -63,144 +56,134 @@ describe('Permit2', () => {
     cy.hardhat().then((hardhat) => {
       const tokenApprovalStub = cy.stub(hardhat.wallet, 'sendTransaction')
       tokenApprovalStub.rejects(USER_REJECTION) // reject token approval
-
       const permitApprovalStub = cy.stub(hardhat.provider, 'send')
       permitApprovalStub.withArgs('eth_signTypedData_v4').rejects(USER_REJECTION) // reject permit approval
       permitApprovalStub.callThrough() // allows non-eth_signTypedData_v4 send calls to return non-stubbed values
 
-      // Clicking the approve button should trigger a token approval that will be rejected by the user (tokenApprovalStub).
-      cy.get(APPROVE_BUTTON).click()
+      initiateSwap()
 
-      // The swap component should prompt approval again.
-      cy.get(APPROVE_BUTTON)
-        .should('have.text', `Approve use of ${INPUT_TOKEN.symbol}`)
+      // tokenApprovalStub should reject here, and the modal should revert to the review state.
+      cy.contains('Review Swap')
+        .should('be.visible')
         .then(() => {
-          tokenApprovalStub.restore() // allow token approval
-
           // The user is now allowing approval, but the permit2 signature will be rejected by the user (permitApprovalStub).
-          cy.get(APPROVE_BUTTON).click()
-          cy.get(APPROVE_BUTTON)
-            .should('have.text', `Approve use of ${INPUT_TOKEN.symbol}`)
+          tokenApprovalStub.restore() // allow token approval
+          cy.get(getTestSelector('confirm-swap-button')).click()
+          cy.get(getTestSelector('PendingModalContent-title')).should('have.text', 'Approve permit')
+          cy.contains('Approved').should('exist')
+          // permitApprovalStub should reject here, and the modal should revert to the review state.
+          cy.contains('Review Swap')
+            .should('be.visible')
             .then(() => {
               permitApprovalStub.restore() // allow permit approval
+              cy.get(getTestSelector('confirm-swap-button')).click()
 
               // The swap should now be able to proceed, as the permit2 signature will be accepted by the user.
-              cy.get(APPROVE_BUTTON)
-                .click()
-                .then(() => {
-                  const approvalTime = Date.now()
-                  cy.get(APPROVE_BUTTON).should('have.text', 'Approval pending')
+              const approvalTime = Date.now()
 
-                  // There should be a successful Approved notification.
-                  cy.contains('Approved').should('exist')
+              cy.get(getTestSelector('PendingModalContent-title')).should('have.text', 'Confirm Swap')
+              cy.contains('Swapped').should('exist')
 
-                  swaps()
-
-                  expectTokenAllowanceForPermit2ToBeMax()
-                  expectPermit2AllowanceForUniversalRouterToBeMax(approvalTime)
-                })
+              expectTokenAllowanceForPermit2ToBeMax()
+              expectPermit2AllowanceForUniversalRouterToBeMax(approvalTime)
```

Comment:
This is the only section with a variable that needs to be in closure (`approvalTime`), so you should start with a `cy.then` here to avoid nesting beforehand.